File: src/main/scala/com/github/learningwords/exception/BadRequestException.java
package com.github.learningwords.exception;

/**
 * Created by dmgcodevil on 1/17/2015.
 */
public class BadRequestException extends RuntimeException {

    public BadRequestException() {
    }

    public BadRequestException(String detailMessage) {
        super(detailMessage);
    }

    public BadRequestException(String detailMessage, Throwable throwable) {
        super(detailMessage, throwable);
    }

    public BadRequestException(Throwable throwable) {
        super(throwable);
    }
}

File: src/main/scala/com/github/learningwords/exception/BadRequestException.scala
package com.github.learningwords.exception

/**
 * Created by dmgcodevil on 1/17/2015.
 */
class BadRequestException(msg: String) extends RuntimeException(msg) {}

object BadRequestException {

  def apply(): BadRequestException = new BadRequestException(null)

  def create(msg: String): BadRequestException = new BadRequestException(msg)

  def create(msg: String, cause: Throwable) = new BadRequestException(msg).initCause(cause)
}